blaze/expr/tests/test_collections.py

```diff
@@ -108,13 +108,21 @@ def test_join_option_types():
     assert join(b, a, 'x').dshape == dshape('var * {x: int}')
 
 
-def test_join_mismatched_schema():
+def test_join_exceptions():
+    'mismatched schema; no shared fields'
```

Comment:
> How do you feel about testing the contents of the error strings?
> something like:
> ```
> with pytest.raises(Error) as e:
>    ...
> ```

Instead of doing an exact match would it suffice to check substring? This will keep it concise and not require string formatting. I don't think exact match buys us much more once we've validated the correct exception was raised.

```
with pytest.raises(TypeError) as excinfo:
    join(a, b, 'x')

assert 'Schemata of joining columns do not match,' in str(excinfo.value)
```

The exact message for this exception is:
`
'Schemata of joining columns do not match,'
                ' no promotion found for %s=%s and %s=%s' % (
                    on_left[n], left_types[n], on_right[n], right_types[n],
                )
`